pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/ClusterIntegrationTestUtils.java

```diff
@@ -833,6 +700,33 @@ static void testSqlQuery(String pinotQuery, String brokerUrl, org.apache.pinot.c
       return;
     }
 
+    BrokerRequest brokerRequest =
+        PinotQueryParserFactory.get(CommonConstants.Broker.Request.SQL).compileToBrokerRequest(pinotQuery);
+    // Add order by columns which are not in selection clause for comparison purpose
+    List<String> orderByColumns = new ArrayList<>();
+    List<String> selectionColumns = new ArrayList<>();
+    if (isSelectionQuery(brokerRequest) && brokerRequest.getOrderBy() != null && brokerRequest.getOrderBy().size() > 0) {
+      orderByColumns.addAll(CalciteSqlParser.extractIdentifiers(brokerRequest.getPinotQuery().getOrderByList(), false));
+      selectionColumns.addAll(CalciteSqlParser.extractIdentifiers(brokerRequest.getPinotQuery().getSelectList(), false));
+      convertToUpperCase(orderByColumns);
+      convertToUpperCase(selectionColumns);
+
+      if (!selectionColumns.containsAll(orderByColumns)) {
+        List<String> inputRequests = new ArrayList<>();
+        inputRequests.add(pinotQuery);
+        inputRequests.addAll(sqlQueries);
+
+        Set<String> orderByColumnsExcluded = new HashSet<>(orderByColumns);
+        orderByColumnsExcluded.removeAll(selectionColumns);
+
+        // Append order-by columns not in selection clause so the order of query responses can be verified
+        // e.g. we can't verify the order of query `SELECT firstName FROM mytable ORDER BY lastName' if there are duplicate lastNames
```

Comment:
If the values of lastName is exhausted, we can still verify the values, right?

Reply:
Yes, the values are checked. Adding the order-by columns will help to verify the order, the values are still checked by comparing hashed row records.

Reply:
Can you post the sample query that you are trying to solve from order-by clause append?

IMO, the queries from the sample set should have a deterministic result instead of implicitly manipulating the order by clause here.

We cannot do the result comparison check if the query result by definition can be changed (e.g. selection with no order by)

Reply:
I think the problem here is whether we want to check the returned order of selection queries which order-by columns are not in selection clause. We skipped comparison for this case for previous sql comparison, and current pql logic will not verify the order but only compare values. So without the order-by clause append, the sql comparison will be consistent with current pql logic, and all test cases should pass.

Reply:
I'll remove this to be consistent with pql comparison, the values are still being checked.